src/include/sof/audio/igo_nr/igo_nr_comp.h

```diff
@@ -0,0 +1,48 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2021 Intelligo Corporation. All rights reserved.
+ *
+ * Author: Fu-Yun TSUO <fy.tsuo@intelli-go.com>
+ */
+
+#ifndef __SOF_AUDIO_IGO_NR_CONFIG_H__
+#define __SOF_AUDIO_IGO_NR_CONFIG_H__
+
+#include <sof/platform.h>
+#include <sof/audio/audio_stream.h>
+#include <sof/audio/igo_nr/igo_lib.h>
+
+/* IGO_NR only support XCC HIFI3 */
+#if defined __XCC__ && XCHAL_HAVE_HIFI3 == 1
+#include <xtensa/config/core-isa.h>
+#else
+#error "No XCC/HIFI3 found. Cannot build IGO_NR module."
+#endif
+
+#define IGO_FRAME_SIZE (256)
+#define IGO_NR_IN_BUF_LENGTH (IGO_FRAME_SIZE)
+#define IGO_NR_OUT_BUF_LENGTH (IGO_FRAME_SIZE)
+#define IGO_MEM_SIZE         (40 * 1024)
+
+/* IGO_NR component private data */
+struct igo_nr_comp_data {
```

Comment:
do you need this anywhere else other than your c file? if not, we should move it there to make sure it is truly private

Reply:
`IGO_MEM_SIZE`, or more specifically, `uint8_t igo_heap[IGO_MEM_SIZE];` is a workaround because we failed to allocated 40KB memory on heap. The heap on SOF seems far less than 40KB.

Reply:
not igo memsize, im referring to `igo_nr_comp_data` also please rename to just `comp_data` that way it will error out in the compiler if it leaks scope into another component

Reply:
Ah thanks. Fixed now. We took tdfb as reference and didn't know there is a such checking mechanism.